File: backend/model/stock.go
// Package model 定义数据库表对应的结构体。
//
// Go 的 ORM 不是主流方式，这里用 sqlx 的 struct tag 直接映射列名。
// `db:"column_name"` 告诉 sqlx 如何把数据库列映射到 struct 字段。
// `json:"field_name"` 告诉 Gin 如何序列化 JSON。
package model

import "time"

// StockKLine 对应 stock_kline 表。
// 主键：(Code, Scale, TradeDate)
//
// 注意：TradeDate 用 time.Time 是因为 pgx 驱动把 PostgreSQL DATE 类型扫成 time.Time。
// API 返回时在 handler 层格式化为 "2006-01-02"。
type StockKLine struct {
	Code      string    `db:"code"       json:"code"`       // 股票/指数代码
	Scale     string    `db:"scale"      json:"scale"`      // 1d/1w/1mon
	TradeDate time.Time `db:"trade_date" json:"-"`          // 交易日期（pgx 扫出来是 time.Time, json:"-" 不直接暴露）
	Open      float64   `db:"open"       json:"open"`       // 开盘价
	High      float64   `db:"high"       json:"high"`       // 最高价
	Low       float64   `db:"low"        json:"low"`        // 最低价
	Close     float64   `db:"close"      json:"close"`      // 收盘价
	Volume    int64     `db:"volume"     json:"volume"`     // 成交量（股）
	Amount    float64   `db:"amount"     json:"amount"`     // 成交额（元），新浪日线不返回，需东财补充
	Turnover  float64   `db:"turnover"   json:"turnover"`   // 换手率（%），同上
	PctChange float64   `db:"pct_change" json:"pct_change"` // 涨跌幅（%），本地计算
	ChangeAmt float64   `db:"change_amt" json:"change_amt"` // 涨跌额，本地计算
	Amplitude float64   `db:"amplitude"  json:"amplitude"`  // 振幅（%），本地计算
	CreatedAt time.Time `db:"created_at" json:"-"`          // 入库时间（不暴露给前端）
}

// StockInfo 对应 stock_info 表。
type StockInfo struct {
	Code      string    `db:"code"       json:"code"`      // 6 位代码
	Name      string    `db:"name"       json:"name"`      // 股票名称
	Type      string    `db:"type"       json:"type"`      // stock/index
	Market    string    `db:"market"     json:"market"`    // sh/sz/bj
	Board     string    `db:"board"      json:"board"`     // main/chinext/star
	Industry  string    `db:"industry"   json:"industry"`  // 行业分类
	IsActive  bool      `db:"is_active"  json:"is_active"` // 是否活跃
	UpdatedAt time.Time `db:"updated_at" json:"-"`         // 更新时间
}

File: front/types/stock.ts
export interface StockInfo {
  code: string
  name: string
  type: string
  market: string
  board: string
  industry: string
  is_active: boolean
}

export interface StockQuote {
  code: string
  name: string
  open: number
  prev_close: number
  price: number
  high: number
  low: number
  volume: number
  amount: number
  date: string
  time: string
  turnover: number
  pct_change: number
}

export interface KlinePoint {
  time: string
  open: number
  high: number
  low: number
  close: number
  volume: number
  amount?: number
  turnover?: number
  pct_change?: number
}

/** 当日分时图数据点（分钟级） */
export interface MinutePoint {
  /** 时间，如 "09:30"（东财/腾讯）或 "2026-08-19 09:31"（Yahoo epoch 转本地） */
  time: string
  /**
   * 完整时间戳 "YYYY-MM-DD HH:mm"（源自带日期时保留，如东财 "2026-08-19 09:30"、
   * Yahoo epoch 转本地；腾讯 HHmm 无日期信息则缺省）。
   * 供触摸浮层十字光标展示「年月日 时分」，跨零点时段（美股 21:30→04:00）逐点准确。
   */
  timeFull?: string
  /** 现价 */
  price: number
  /** 均价（累计成交额/累计成交量）；源未提供时为空 */
  avg: number | null
  /** 该分钟成交量 */
  volume: number
  amount?: number
}

/** 当日分时图数据（含昨收基准） */
export interface MinuteResult {
  /** 昨收（分时基准线），源未提供时为空 */
  preClose: number | null
  /**
   * 昨结算（期货口径的涨跌幅基准，如 SHFE 沪主连 / COMEX）：
   * 期货的「昨收」是昨日最后成交价（如 沪银 16771），而涨跌幅按「昨结算」（16611）计算，
   * 两者不同；非期货（股票 / 指数 / 汇率）无此字段或为 0。
   */
  preSettlement?: number | null
  points: MinutePoint[]
  /** 源返回的证券名（如东财美股「英伟达」），供代理股合成页标注中文名 */
  name?: string
}

export interface KlineResult {
  code: string
  scale: string
  klines: KlinePoint[]
  count: number
}

export interface SectorBoard {
  plate_code: string
  plate_name: string
  cid: number
}

export interface SectorMember extends StockInfo {
  price?: number
  pct_change?: number
}

export interface NewsItem {
  id?: string
  title: string
  url: string
  source?: string
  time?: string
  summary?: string
}

export interface NewsListResponse {
  code?: string
  page?: number
  size?: number
  count: number
  hasMore?: boolean
  news: NewsItem[]
}

export interface AnnouncementItem {
  id: string
  title: string
  url: string
  time: string
  pdf?: string
}

export interface AnnouncementListResponse {
  code: string
  page: number
  count: number
  items: AnnouncementItem[]
}

export interface SectorMembersResponse {
  cid: number
  count: number
  stocks: string[]
}
